### DISCUSSION TITLE
Using key function with typescript

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#659] [General] [answered]
The TS compiler is complaining about a possible undefined variable when using the key callback pattern:

```ts
const { session } = useSession()
useSWR(() => ["/v2/order/recipe/summary", session.id],
```

This will complain that `session` might be undefined. Since this is intentional what is the swr teams recommendation for getting rid of the warning? Disable it via comment?

**Comment by sergiodxa** [comment, marked as answer]
You can use a condition
```ts
const { session } = useSession()
useSWR(() => session?.id ? ["/v2/order/recipe/summary", session.id] : null, fetcher);
```
This way, SWR will not run the fetch until session is defined which will prevent possible failing requests.

**Reply by pke** [reply]
Yes, that's my current approach already. I just wondered about the callback approach.
